pkg/controller/common/association/rbac.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package association
+
+import (
+	"time"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/rbac"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Unbinder interface {
+	Unbind(associated commonv1.Associated) error
+}
+
+// CheckAndUnbind checks if a reference is allowed and unbind the association if it is not the case
+func CheckAndUnbind(
+	accessReviewer rbac.AccessReviewer,
+	associated commonv1.Associated,
+	object runtime.Object,
```

Comment:
`object` is not very explicit: is it the src object holding the reference, or the referenced one?
Let's maybe rename it to `remoteObject` or `referencedObject`.